src/node_platform.cc

```diff
@@ -100,8 +103,19 @@ class WorkerThreadsTaskRunner::DelayedTaskScheduler {
   static void FlushTasks(uv_async_t* flush_tasks) {
     DelayedTaskScheduler* scheduler =
         ContainerOf(&DelayedTaskScheduler::loop_, flush_tasks->loop);
-    while (std::unique_ptr<Task> task = scheduler->tasks_.Pop())
+
+    std::vector<std::unique_ptr<Task>> tasks_to_run;
+    {
+      auto locked = scheduler->tasks_.Lock();
+      std::unique_ptr<Task> task;
+      while ((task = locked.Pop())) {
+        tasks_to_run.push_back(std::move(task));
+      }
+    }
```

Comment:
```suggestion
    auto tasks_to_run = scheduler->tasks_.Lock().PopAll();
```
That's equivalent, right?